network_wrangler/scenario.py

```diff
@@ -604,224 +593,288 @@ def apply_projects(self, project_list: Collection[str]):
-    transit_dir: optional
-        path to base scenario transit files
-    validate:
-        boolean indicating whether to validate the base network or not
+    roadway: Optional[dict] = None,
+    transit: Optional[dict] = None,
+    applied_projects: Optional[list] = [],
+    conflicts: Optional[list] = [],
+    config: WranglerConfig = DefaultConfig,
+) -> dict:
+    """Creates a base scenario dictionary from roadway and transit network files.
+
+    Args:
+        roadway: kwargs for load_roadway_from_dir
+        transit: kwargs for load_transit from dir
+        applied_projects: list of projects that have been applied to the base scenario.
+        conflicts: list of conflicts that have been identified in the base scenario.
+        config: WranglerConfig instance.
     """
-    if roadway_dir:
-        base_network_shape_file = os.path.join(roadway_dir, base_shape_name)
-        base_network_link_file = os.path.join(roadway_dir, base_link_name)
-        base_network_node_file = os.path.join(roadway_dir, base_node_name)
+    if roadway:
+        road_net = load_roadway_from_dir(**roadway, config=config)
     else:
-        base_network_shape_file = base_shape_name
-        base_network_link_file = base_link_name
-        base_network_node_file = base_node_name
-
-    road_net = RoadwayNetwork.read(
-        link_file=base_network_link_file,
-        node_file=base_network_node_file,
-        shape_file=base_network_shape_file,
-        fast=not validate,
-    )
+        road_net = None
+        WranglerLogger.info(
+            "No roadway directory specified, base scenario will have empty roadway network."
+        )
 
-    if transit_dir:
-        transit_net = TransitNetwork.read(transit_dir)
+    if transit:
+        transit_net = load_transit(**transit, config=config)
+        if roadway:
+            transit_net.road_net = road_net
     else:
         transit_net = None
         WranglerLogger.info(
             "No transit directory specified, base scenario will have empty transit network."
         )
 
-    transit_net.set_roadnet(road_net, validate_consistency=validate)
-    base_scenario = {"road_net": road_net, "transit_net": transit_net}
+    base_scenario = {
+        "road_net": road_net,
+        "transit_net": transit_net,
+        "applied_projects": applied_projects,
+        "conflicts": conflicts,
+    }
 
     return base_scenario
+
+
+def extract_base_scenario_metadata(base_scenario: dict) -> dict:
+    """Extract metadata from base scenario rather than keeping all of big files.
+
+    Useful for summarizing a scenario.
+    """
+    _skip_copy = ["road_net", "transit_net", "config"]
+    out_dict = {k: v for k, v in base_scenario.items() if k not in _skip_copy}
+    if isinstance(base_scenario.get("road_net"), RoadwayNetwork):
+        nodes_file = Path(base_scenario["road_net"].nodes_df.attrs["source_file"])
+        out_dict["roadway"] = {"dir": str(nodes_file.parent), "suffix": str(nodes_file.suffix)}
```

Comment:
Can't get the `nodes_file` path from nodes_df. The nodes_df.attrs is updated with the default settings [here](https://github.com/wsp-sag/network_wrangler/blob/selection-refactor/network_wrangler/roadway/links/io.py#L49-L50). Can we use links_df instead?

Reply:
There was a bug in the `validate_to_df()` that forgot to copy the `attr` properties at validation (from all the merging) that I fixed - can you confirm that it works now? 

WRT resolution, we shouldn't just change from nodes-->links. Need to fix if the `attr` is dropped.